jib-core/src/main/java/com/google/cloud/tools/jib/event/events/TimerEvent.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2018 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.event.events;
+
+import com.google.cloud.tools.jib.event.JibEvent;
+import java.time.Duration;
+import java.util.Optional;
+
+/**
+ * Timer event for timing various part of Jib's execution.
+ *
+ * <p>Timer events follow a specific {@link Timer} through a {@link State#START}, {@link
+ * State#IN_PROGRESS}, and {@link State#FINISHED} states. The duration indicates the duration since
+ * the last {@link TimerEvent} emitted for the {@link Timer}.
+ *
+ * <p>Timers can also define a hierarchy.
+ */
+public class TimerEvent implements JibEvent {
+
+  /** The state of the timing. */
+  public enum State {
+
+    /** The timer has started timing. Duration will be 0. */
+    START,
+
+    /** The timer is still in progress. Duration captures the time since the last event. */
+    IN_PROGRESS,
+
+    /** The timer has finished timing. Duration captures the time since the last event. */
```

Comment:
I think it makes more sense to return the total time.  A swimmer's race records the individual lap times and their total time.  Or maybe TimerEvent should carry the duration and elapsed (total)

Reply:
Made TimerEvent carry the duration and elapsed